### PR TITLE
PIM-4891 : check by code if a locale is activated

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                  | A
| ------------------ | ---
| Migration script?  |n
| Behats             |y
| Specs & Static     |y
| Changelog updated  |n

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php [hunk 1]
```diff
@@ -79,10 +79,39 @@ public function getLocaleByCode($code)
...
+
+    /**
+     * Check if a locale is activated
+     *
+     * @param LocaleInterface[] $locales
+     * @param string            $localeCode
+     *
+     * @return bool
+     */
+    public function isActivatedLocale(array $locales, $localeCode)
```
Comment (ClementGautier) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: isLocaleActivated ?

**Comment by jjanvier** [comment]
gtm

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php (hunk 1 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: +1

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php [hunk 2]
```diff
@@ -79,10 +79,39 @@ public function getLocaleByCode($code)
...
+    {
+        $foundLocale = null;
+
+        $iterator = new \ArrayIterator($locales);
```
Comment (ClementGautier) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: Why do you use an Iterator here ? You can do the same thing with an array. You can write this function that way for example:
```php
public function isLocaleActivated(array $locales, $code)
{
    foreach ($locales as $locale) {
        if ($code === $locale->getCode()) {
            return $locale->isActivated();
        }
    }

    throw new \OutOfBoundsException(sprintf('locale code %s is unknown', $code));
}
```

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php [hunk 3]
```diff
@@ -79,10 +79,39 @@ public function getLocaleByCode($code)
...
+        while ($iterator->valid() && null === $foundLocale) {
+            $locale = $iterator->current();
+            if ($locale->getCode() === $localeCode) {
+                $foundLocale = $locale;
+            }
+            $iterator->next();
+        }
+
+        if (null === $foundLocale) {
+            throw new \OutOfBoundsException(sprintf('locale code %s is unknown', $localeCode));
```
Comment (ClementGautier) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: Not sure that `OutOfBoundsException` if the exception you want. I would prefer a `LogicException`. `OutOfBoundException` are thrown if a value is not a valid key, it represents errors that cannot be detected at compile time.

**Comment by ClementGautier** [comment]
GTM after comments fixed

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php (hunk 2 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: +1 will be more readable

Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: :+1:

File: src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php (hunk 3 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: :+1:

Comment (jmleroux) [inline src/Pim/Bundle/CatalogBundle/Manager/LocaleManager.php]: @ClementGautier not a `LogicException` but ok for a `RuntimeException`

**Pull request merged**